components/locale_canonicalizer/src/provider.rs

```diff
@@ -7,13 +7,33 @@
 //! Read more about data providers: [`icu_provider`]
 
 use icu_locid::LanguageIdentifier;
-use tinystr::TinyStr4;
+use tinystr::{TinyStr4, TinyStr8};
 
 pub mod key {
     use icu_provider::{resource_key, ResourceKey};
+    pub const ALIASES_V1: ResourceKey = resource_key!(aliases, "aliases", 1);
     pub const LIKELY_SUBTAGS_V1: ResourceKey = resource_key!(likelysubtags, "likelysubtags", 1);
 }
 
+#[derive(Debug, PartialEq, Clone, Default)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct AliasesV1 {
```

Comment:
Thought: These two data structs will need to be migrated to ZeroVec, but that cannot be done until #667 is done.  I am working on #667 now, so we should coordinate this.